backend-es/test/snapshots-out/Snapshot.GenericBounded04.js

```diff
@@ -0,0 +1,189 @@
+          }
+          $runtime.fail();
+        }
+        $runtime.fail();
+      },
+      Eq0: () => eqTest42
+    };
+  };
+};
+const genericTest4 = {
+  to: x => {
+    if (x.tag === "Inl") { return $Test4("Bottom4", x._1._1, x._1._2._1, x._1._2._2._1, x._1._2._2._2._1, x._1._2._2._2._2._1, x._1._2._2._2._2._2._1, x._1._2._2._2._2._2._2); }
+    if (x.tag === "Inr") {
+      if (x._1.tag === "Inl") { return Ignored4; }
+      if (x._1.tag === "Inr") {
+        return $Test4("Top4", x._1._1._1, x._1._1._2._1, x._1._1._2._2._1, x._1._1._2._2._2._1, x._1._1._2._2._2._2._1, x._1._1._2._2._2._2._2._1, x._1._1._2._2._2._2._2._2);
+      }
+      $runtime.fail();
+    }
+    $runtime.fail();
+  },
+  from: x => {
+    if (x.tag === "Bottom4") {
+      return Data$dGeneric$dRep.$Sum(
+        "Inl",
+        Data$dGeneric$dRep.$Product(
+          x._1,
+          Data$dGeneric$dRep.$Product(
+            x._2,
+            Data$dGeneric$dRep.$Product(x._3, Data$dGeneric$dRep.$Product(x._4, Data$dGeneric$dRep.$Product(x._5, Data$dGeneric$dRep.$Product(x._6, x._7))))
+          )
+        )
+      );
+    }
+    if (x.tag === "Ignored4") { return Data$dGeneric$dRep.$Sum("Inr", Data$dGeneric$dRep.$Sum("Inl", Data$dGeneric$dRep.NoArguments)); }
+    if (x.tag === "Top4") {
+      return Data$dGeneric$dRep.$Sum(
+        "Inr",
+        Data$dGeneric$dRep.$Sum(
+          "Inr",
+          Data$dGeneric$dRep.$Product(
+            x._1,
+            Data$dGeneric$dRep.$Product(
+              x._2,
+              Data$dGeneric$dRep.$Product(x._3, Data$dGeneric$dRep.$Product(x._4, Data$dGeneric$dRep.$Product(x._5, Data$dGeneric$dRep.$Product(x._6, x._7))))
+            )
+          )
+        )
+      );
+    }
+    $runtime.fail();
+  }
+};
+const boundedTest4 = dictOrd1 => {
+  const ordTest41 = ordTest4(dictOrd1);
+  return dictBounded => dictBounded1 => {
+    const ordTest42 = ordTest41(dictBounded1.Ord0());
+    return {
+      bottom: $Test4(
+        "Bottom4",
+        -2147483648,
+        {a: -2147483648},
+        Snapshot$dGenericBounded01.Bottom1,
+        Snapshot$dGenericBounded02.boundedTest2.bottom,
+        Snapshot$dGenericBounded03.boundedTest3.bottom,
+        dictBounded1.bottom,
+        dictBounded.bottom
+      ),
+      top: $Test4(
+        "Top4",
+        2147483647,
+        {a: 2147483647},
+        Snapshot$dGenericBounded01.Top1,
+        Snapshot$dGenericBounded02.boundedTest2.top,
+        Snapshot$dGenericBounded03.boundedTest3.top,
+        dictBounded1.top,
+        dictBounded.top
+      ),
+      Ord0: () => ordTest42
+    };
+  };
+};
+const boundedTest41 = /* #__PURE__ */ boundedTest4(/* #__PURE__ */ Data$dTuple.ord1Tuple(Data$dOrd.ordInt))(/* #__PURE__ */ Data$dTuple.boundedTuple(Data$dBounded.boundedInt)(Data$dBounded.boundedInt))(Data$dBounded.boundedInt);
+const test5 = /* #__PURE__ */ (() => boundedTest41.bottom)();
+const test6 = /* #__PURE__ */ (() => boundedTest41.top)();
```

Comment:
I'm not sure why these aren't getting outputted as just `boundedTest41.bottom`/`top`

Reply:
Because property access in JS is not necessarily pure, they could be getters. Therefore it needs the wrapper for the PURE annotation.

Reply:
Gotcha. Thanks for explaining.